app/src/lib/stores/app-store.ts

```diff
@@ -3417,19 +3426,150 @@ export class AppStore extends TypedBaseStore<IAppState> {
     return this._refreshRepository(repository)
   }
 
+  /** This shouldn't be called directly. See `Dispatcher`. */
+  public _setRebaseProgressFromState = async (repository: Repository) => {
```

Comment:
**Non-blocking**
Are there any benefits to defining the function in this way in this class? Only asking because it's the only function in this class that is defined as a lambda expression.

Reply:
Arrow functions inside classes (at least as part of ES2017 support we're targeting in Desktop) are used to ensure that `this` is correctly bound when the function is invoked. In the most part this is unnecessary for classes, but if you find yourself passing a class function for use elsewhere (for example, into a React component) you should ensure it is bound correctly.

We already have a decent set of lint rules to catch these usages, e.g. https://github.com/desktop/desktop/pull/7129#discussion_r267523660 so I'm fine to clean this up because it's not important.

For more details, here's an example from [this blog post](https://medium.com/@charpeni/arrow-functions-in-class-properties-might-not-be-as-great-as-we-think-3b3551c440b1) that talks about the tradeoffs:

```ts
class A {
  static color = "red";
  counter = 0;
  
  handleClick = () => {
    this.counter++;
  }
  
  handleLongClick() {
    this.counter++;
  }
}
```

 As part of the transpilation to JS it moves this inside inside the constructor to confirm it's an instance function:

```js
class A {
  constructor() {
    this.counter = 0;

    this.handleClick = () => {
      this.counter++;
    };
  }
  
  handleLongClick() {
    this.counter++;
  }
}
A.color = "red";
```

Reply:
Tidied up in b422d8d22f91867382b810ad2ab51a713e2f1721